vaadin-dialog-flow-parent/vaadin-dialog-flow/src/test/java/com/vaadin/flow/component/dialog/DialogOpenedChangeListenerTest.java

```diff
@@ -0,0 +1,97 @@
+package com.vaadin.flow.component.dialog;
+
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.server.VaadinSession;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public class DialogOpenedChangeListenerTest {
```

Comment:
Since the new implementation uses custom logic to dispatch the opened change event, added a new set of tests to verify that they are properly fired, that the is from client flag is correct, and that no duplicate events are fired.